### ISSUE TITLE
[Android] [Xiaomi] Fatal Exception on all Xiaomi Devices

### ISSUE THREAD

**Issue by claudius-kienle** [flutter/flutter#111178]
I just released my flutter based app as replacement for the native implementations on android. 
Since then, Firebase Crashlytics reports a lot of fatal crashes directly after application startup, but only on Xiaomi devices (POCO F1, POCO F3, Redmi Note 9 Pro, ...).

Sadly, the stacktrace of the crashes does not contain enough information for me to debug.

The stack trace is following:
```bash
StandardMethodCodec.decodeEnvelope

Fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: PlatformException(error, Unsupported value: 'UserHandle{0}' of type 'class android.os.UserHandle', null, null). Error thrown Instance of 'ErrorDescription'.
       at StandardMethodCodec.decodeEnvelope(message_codecs.dart:607)
       at MethodChannel._invokeMethod(platform_channel.dart:167)
       at EventChannel.receiveBroadcastStream.<fn>(platform_channel.dart:506)
```

flutter docker -v
```bash
[✓] Flutter (Channel stable, 3.0.4, on Manjaro Linux 5.10.136-1-MANJARO, locale C.UTF-8)
    • Flutter version 3.0.4 at /home/user/.development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 85684f9300 (2 months ago), 2022-06-30 13:22:47 -0700
    • Engine revision 6ba2af10bb
    • Dart version 2.17.5
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/user/Android/Sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Android Studio (version 2021.2)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 67.1.2
    • Dart plugin version 212.5744
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

Does someone have a suggestion on how to debug this issue?

**Comment by exaby73** [comment]
Hello @claudius-kienle. Thank you for filing this issue. Labelling for further insights from the team.

**Comment by chinmaygarde** [comment]
Without a reproducible test case, we cannot make any progress.

From Triage: If we had to guess, it could be a platform plugin trying to serialize a [`android.os.UserHandle`](https://developer.android.com/reference/android/os/UserHandle) in a platform message. How and where that happens, we cannot say without a test case. I checked the engine source code and it isnt because of an built-in plugin.
